### PR TITLE
Add username/password support for MQTT

### PR DESCRIPTION
**Pull request by basyskom-jvoe**

### CODE REVIEW

File: plugins/mqtt/ua_mqtt_adapter.c [hunk 1]
```diff
@@ -130,9 +130,38 @@ connectMqtt(UA_PubSubChannelDataMQTT* channelData){
...
+            return UA_STATUSCODE_BADOUTOFMEMORY;
+        }
+        memcpy(username, channelData->mqttUsername->data, channelData->mqttUsername->length);
+    }
+
+    if (channelData->mqttPassword) {
+        /* Convert password UA_String to char* null terminated */
+        password = (char*)calloc(1,channelData->mqttPassword->length + 1);
+        if(!password){
+            UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "PubSub MQTT: Connection creation failed. Out of memory.");
```
Comment (jpfr) [inline plugins/mqtt/ua_mqtt_adapter.c:152]: Add `UA_free(username)` here.

File: plugins/ua_network_pubsub_mqtt.h [hunk 2]
```diff
@@ -25,12 +25,12 @@ typedef struct {
     UA_Connection *connection;
     void * mqttClient;
     void (*callback)(UA_ByteString *encodedBuffer, UA_ByteString *topic);
+    UA_String *mqttUsername;
```
Comment (jpfr) [inline plugins/ua_network_pubsub_mqtt.h:28]: Make these non-pointer UA_String.
And copy the content over.
So we don't keep a dependency on the connectionConfig lifecycle.

Comment (basyskom-jvoe) [reply plugins/ua_network_pubsub_mqtt.h]: Thanks for the review!

I hope I have covered all places where the two strings have to be cleared.

**Comment by jpfr** [comment]
Rebased for the CI.

**Comment by jpfr** [comment]
Great! Merged. 👍 

I think using logins with MQTT will be a frequent question.
Can you add this to the API documentation and/or existing or new tutorials?

**Comment by basyskom-jvoe** [comment]
Thanks for the merge!
I have extended the example in #3644

**Pull request merged**